int32 machine_num
int32[] transport_durations # size should be equal to number of machines